java/code/src/com/suse/manager/hub/HubManager.java

```diff
@@ -1147,6 +1160,281 @@ private IssV3ChannelResponse buildIssV3ChannelResponse(Channel channel) {
+        if (parentLabel == null) {
+            return false;
+        }
+        // Check if parent is synced
+        if (syncedLabels.contains(parentLabel)) {
+            return true;
+        }
+        // Check parent's ancestors recursively
+        return hasAncestorSynced(parentLabel, syncedLabels, childToParentMap);
+    }
+
+
+    /**
+     * Sync the channels from "this" hub to the selected peripheral
+     * @param user the SatAdmin
+     * @param peripheralId the peripheral id
+     * @param orgId the org id to sync the channels to
+     * @param channelsLabels the list of channels labels from the hub
+     */
+    public void syncChannelsByLabelForPeripheral(User user, Long peripheralId, Long orgId, List<String> channelsLabels)
+            throws CertificateException, IOException {
+        ensureSatAdmin(user);
+        // Get peripheral and prepare client
+        IssPeripheral peripheral = hubFactory.findPeripheralById(peripheralId);
+        HubInternalClient client = createClientForPeripheral(peripheral);
+        // Prepare channels for synchronization
+        Set<Long> syncedChannelIds = getSyncedChannelIds(peripheral);
+        List<Channel> channelsToSync = prepareChannelsToSync(channelsLabels, syncedChannelIds);
+        // Execute the synchronization
+        synchronizeChannels(peripheral, channelsToSync, syncedChannelIds, orgId, client);
+    }
+
+    /**
+     * Create a client for communicating with the peripheral
+     */
+    private HubInternalClient createClientForPeripheral(IssPeripheral peripheral) throws CertificateException {
+        IssAccessToken accessToken = hubFactory.lookupAccessTokenFor(peripheral.getFqdn());
+        return clientFactory.newInternalClient(
+                peripheral.getFqdn(),
+                accessToken.getToken(),
+                peripheral.getRootCa()
+        );
+    }
+
+    /**
+     * Get the set of channel IDs that are already synced to the peripheral
+     */
+    private Set<Long> getSyncedChannelIds(IssPeripheral peripheral) {
+        return peripheral.getPeripheralChannels().stream()
+                .map(pc -> pc.getChannel().getId())
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Prepare the list of channels to sync, including originals and parent channels
+     */
+    private List<Channel> prepareChannelsToSync(List<String> channelsLabels, Set<Long> syncedChannelIds) {
+        Set<Channel> requestedChannels = loadChannelsByLabel(channelsLabels);
+        Set<Channel> completeChannelSet = ensureParentChildHierarchy(requestedChannels, syncedChannelIds);
+        return sortChannelsByHierarchy(completeChannelSet);
+    }
+
+    /**
+     * Load channels by their IDs
+     */
+    private Set<Channel> loadChannelsByLabel(List<String> channelsLabels) {
+        List<Long> channelsIds = ChannelFactory.getChannelIds(channelsLabels);
+        return new HashSet<>(ChannelFactory.getSession()
```

Comment:
```suggestion
        return new HashSet<>(HibernateFactory.getSession()
```

and add an import for HibernateFactory.
Fixes a Sonarcloud issue

Reply:
For this I have pushed a fix